library/include/modules/Buildings.h

```diff
@@ -237,14 +237,20 @@ DFHACK_EXPORT std::string getRoomDescription(df::building *building, df::unit *u
  * starting at the top left and moving right, row by row,
  * the block's items are checked for anything on the ground within that stockpile.
  */
-class DFHACK_EXPORT StockpileIterator : public std::iterator<std::input_iterator_tag, df::item>
```

Comment:
Why did this need to change? New version seems harder to maintain to me.

Reply:
iterator subclasses are deprecated. I had to migrate to the new way of doing things.
https://www.fluentcpp.com/2018/05/08/std-iterator-deprecated/

Reply:
Sounds good. I think I was confusing this with `std::vector::iterator`, which I don't think we can use here.